Firestore/core/src/firebase/firestore/model/field_value.cc

```diff
@@ -55,6 +55,19 @@ bool Comparable(Type lhs, Type rhs) {
   }
 }
 
+// Makes a copy excluding the specified child, which is expected to be assigned
+// different value afterwards.
+ObjectValue::Map MakePartialCopy(const ObjectValue::Map& object_map,
```

Comment:
Nit: Given that this is temporary code that's not yet in use in production this seems like overkill.

Couldn't you just do a copy using the copy constructor and perform whatever mutation is required on the copy?

I only mention this because implementing "erase" with a function named "MakePartialCopy" is not obvious. Perhaps it would make sense if this were "CopyExcept" but I still favor doing the simpler and more obvious thing.

Reply:
Changed name from `MakePartialCopy` to `CopyExcept`. Use `CopyExcept` here instead of the straightforward `operator=` to address reviewer's concern to avoid non-necessary deep copy of child.